Add Teams Team Report (#19900)
microsoftgraph/microsoft-graph-docs: api-reference/beta/api/reportroot-getteamsteamcounts.md

---

> title: "reportRoot: getTeamsTeamCounts"

Reviewer: Please add the method to the methods table of the respective topic,

Author: Added the function to the table

> description: "Get the Teams Team count by Teams type."

Reviewer: We need to make capitalization changes, I suggest:
```suggestion
description: "Get the Teams team count by teams type."
```

Author: Fixed

ms.localizationpriority: medium
ms.prod: "reports"
author: "pranoychaudhuri"
doc_type: apiPageType
---

# reportRoot: getTeamsTeamCounts

Namespace: microsoft.graph

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

> Get the Teams team count by team type.

Reviewer 2: I suggest that we reword this description to avoid using the word "team" three times and help clarify this statement a bit. For example, would this suggested revision be accurate: 

"Get the number of teams of a particular type in an instance of Microsoft Teams."

Also, how do they specify the team "type"?

Author: The type could be 'Private Teams', 'Active Private Teams', 'Public Teams', 'Active Public Teams'

Reviewer 2: @airicaruss , where do we describe this in the docs? Is there a place that we can link to?

> Get the Teams Team count by Teams type.

Reviewer: We need to make capitalization changes, I suggest:
```suggestion
Get the Teams team count by team type.
```

Author: Fixed

| Permission type                        | Permissions (from least to most privileged) |
| :------------------------------------- | :--------------------------------------- |
| Delegated (work or school account)     | Reports.Read.All                         |
| Delegated (personal Microsoft account) | Not supported.                           |
| Application                            | Reports.Read.All                         |

> **Note:** For delegated permissions to allow apps to read service usage reports on behalf of a user, the tenant administrator must have assigned the user the appropriate Azure Active Directory limited administrator role. For more details, see [Authorization for APIs to read Microsoft 365 usage reports](/graph/reportroot-authorization).
## HTTP request

<!-- { "blockType": "ignored" } -->

```http
GET /reports/getTeamsTeamCounts(period='{period_value}')
```

## Function parameters

In the request URL, provide the following parameters with a valid value.

| Parameter | Type   | Description                              |
| :-------- | :----- | :--------------------------------------- |

> | period    | string | Specifies the length of time over which the report is aggregated. The supported values for {period_value} are: `D7`, `D30`, `D90`, and `D180`. These values follow the format D*n* where *n* represents the number of days over which the report is aggregated. Required. |

Reviewer: ```suggestion
| period    | string | Specifies the length of time over which the report is aggregated. The supported values for {period_value} are: `D7`, `D30`, `D90`, and `D180`. These values follow the format D*n*, where *n* represents the number of days over which the report is aggregated. Required. |
```

> **Note:** You need to set either **period** or **date** in the request URL.
## Optional query parameters

This method supports the `$format` [OData query parameter](/graph/query-parameters) to customize the response. The default output type is `text/csv`. However, if you want to specify the output type, you can use the OData `$format` query parameter to set the default output to `text/csv` or `application/json`.

## Request headers

| Name          | Description               |
| :------------ | :------------------------ |
| Authorization | Bearer {token}. Required. |

> ## Response

Reviewer 2: We are missing the required H2 section "Request body". API topics adhere to a defined structure so that users can easily find the information they're looking for. Please add the missing section.

Author: Added

### CSV

> If successful, this method returns a `302 Found` response that redirects to a preauthenticated download URL for the report. That URL can be found in the `Location` header in the response.

Reviewer: ```suggestion
If successful, this method returns a `302 Found` response that redirects to a pre-authenticated download URL for the report. That URL can be found in the `Location` header in the response.
```

> Preauthenticated download URLs are only valid for a short period of time (a few minutes) and do not require an `Authorization` header.

Reviewer: ```suggestion
Pre-authenticated download URLs are only valid for a short period (a few minutes) and do not require an `Authorization` header.
```

---
title: "reportRoot: getTeamsTeamCounts"
description: "Get the number of teams of a particular type in an instance of Microsoft Teams."
ms.localizationpriority: medium
ms.prod: "reports"
author: "pranoychaudhuri"
doc_type: apiPageType
---

# reportRoot: getTeamsTeamCounts

Namespace: microsoft.graph

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

> Get the number of teams of a particular type in an instance of Microsoft Teams.

Reviewer 2: Can you provide a link here to where we list the applicable types of teams?
